### ISSUE TITLE
Bug: z.default() implies not-required in OpenAPI

### ISSUE THREAD

**Issue by kernwig** [asteasolutions/zod-to-openapi#188]
When defining a spec like `z.string().default("active")`, the resulting OpenAPI spec has the default, but also `required: true`. This is confusing to the reader, as it flags an optional input as required when it is not.

Can work around with explicit `z.optional()`, but that is implied by the `z.default()`.

Found when used on a query-string parameter, if that makes a difference.

Errant OpenAPI spec snippet:

```yaml
      parameters:
        - schema:
            type: string
            default: active
          required: true
          name: state
          in: query
```

**Comment by AGalabov** [comment]
@kernwig I think this is a valid point. If you feel like it - please go ahead and open a PR. If not I'll try to get back to this soon.

**Comment by AGalabov** [comment]
@kernwig this should now be fixed and available as part of the [v6.3.0 release](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v6.3.0).

Thank you for reporting!

**Comment by kernwig** [comment]
@AGalabov Thank you for the enhancement/fix!
